File: CS/GaugePropertyEditor.Web/WebGaugePropertyEditorBase.cs
using System;
using System.Drawing;
using DevExpress.Web.ASPxGauges;
using DevExpress.ExpressApp.Web.Editors;

namespace GaugePropertyEditor.Web {
    public abstract class WebGaugePropertyEditorBase : WebPropertyEditor {
        private object bindableComponentCore;
        protected WebGaugePropertyEditorBase(Type objectType, DevExpress.ExpressApp.Model.IModelMemberViewItem info)
            : base(objectType, info) {
        }
        protected override object CreateControlCore() {
            ASPxGaugeControl gaugeControl = new ASPxGaugeControl();
            gaugeControl.BackColor = Color.Transparent;
            SetupGaugeCore(gaugeControl);
            return gaugeControl;
        }
        protected override void OnControlCreated() {
            base.OnControlCreated();
            UpdateEditorState();
        }
        public object BindableComponent {
            get { return bindableComponentCore; }
            set { bindableComponentCore = value; }
        }
        public abstract void SetupGaugeCore(ASPxGaugeControl gaugeContainer);
        protected override System.Web.UI.WebControls.WebControl CreateEditModeControlCore() { return null; }
        protected override void ReadEditModeValueCore() { }
    }
}

File: VB/GaugePropertyEditor.Web/WebGaugePropertyEditorBase.vb
Imports System
Imports System.Drawing
Imports DevExpress.Web.ASPxGauges
Imports DevExpress.ExpressApp.Web.Editors
Imports System.Web.UI.WebControls

Namespace GaugePropertyEditor.Web
	Public MustInherit Class WebGaugePropertyEditorBase
		Inherits WebPropertyEditor

		Private bindableComponentCore As Object
		Protected Sub New(ByVal objectType As Type, ByVal info As DevExpress.ExpressApp.Model.IModelMemberViewItem)
			MyBase.New(objectType, info)
		End Sub
		Private Function CreateGaugeControl() As WebControl
			Dim gaugeControl As New ASPxGaugeControl()
			gaugeControl.BackColor = Color.Transparent
			SetupGaugeCore(gaugeControl)
			Return gaugeControl
		End Function
		Protected Overrides Function CreateViewModeControlCore() As WebControl
			Return CreateGaugeControl()
		End Function
		Protected Overrides Function CreateEditModeControlCore() As WebControl
			Return CreateGaugeControl()
		End Function
		Protected Overrides Sub OnControlCreated()
			MyBase.OnControlCreated()
			UpdateEditorState()
		End Sub
		Public Property BindableComponent() As Object
			Get
				Return bindableComponentCore
			End Get
			Set(ByVal value As Object)
				bindableComponentCore = value
			End Set
		End Property
		Public MustOverride Sub SetupGaugeCore(ByVal gaugeContainer As ASPxGaugeControl)
		Protected Overrides Sub ReadEditModeValueCore()
		End Sub
	End Class
End Namespace